Lib/test/test_threading.py

```diff
@@ -1604,6 +1604,20 @@ def test_interrupt_main_invalid_signal(self):
         self.assertRaises(ValueError, _thread.interrupt_main, signal.NSIG)
         self.assertRaises(ValueError, _thread.interrupt_main, 1000000)
 
+    def test_can_interrupt_tight_loops(self):
+        #Nothing to assert here. It just shouldn't hang.
+
+        cont = True
+        def worker():
+            while cont:
+                pass
+
+        t = threading.Thread(target=worker)
+        t.start()
+        time.sleep(0.1)
+        cont = False
+        t.join()
```

Comment:
Please, add a timeout to the join so if it hangs it doesn't hang the full test suite and decorate the function with `@threading_helper.reap_threads`. Also, ideally we don't use a sleep as a syncronization promitive

Reply:
I don't think the timeout will help in this case. If no other threads can run, then we can never reach the join call.

Reply:
Then add a maximum number of iterations or something. The objective here is that the test suite cannot fully hang if the test fails.